src/common/reducers/entities.js

```diff
@@ -0,0 +1,66 @@
+import merge from 'lodash/merge';
+
+import * as ActionTypes from '../actions/repository';
+
+// export only for testing
+export function repository(state={
+  __isSelected: false,
+  __isFetching: false,
+  __error: null
+}, action) {
+
+  switch(action.type) {
+    case ActionTypes.REPOSITORY_TOGGLE_SELECT: {
+      const wasSelected = state.__isSelected;
+
+      return {
+        ...state,
+        __isSelected: !wasSelected
+      };
+    }
+    case ActionTypes.REPOSITORY_BUILD: {
+      return {
+        ...state,
+        __isFetching: true,
+        __isSelected: true
+      };
+    }
+    case ActionTypes.REPOSITORY_SUCCESS:
```

Comment:
What we're doing here is creating a snap from a given repository, and the request returns a new snap, which we can insert into a refactored snap state and have the correct data on the next view (dashboard).

More accurate name would be:

```
// add a repository to launchpad's build queue, returns a snap representation
const REPOSITORY_ADD = 'REPOSITORY_ADD';
```